specs/create-rbd-spec.adoc

```diff
@@ -0,0 +1,236 @@
+
+* The flow consist of three steps.
+
+* In the first step, user needs to fill below items to the form -
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735641)
+
+    - RBD Name
+
+    - RBDs to create(UI will autogenerate the name of RBDs based on the number)
+
+    - cluster(User will select from the drop down which will list all the clusters present in Tendrl)
+
+    - Target size
+
+* If the target size * RBDs to create > available storage in this cluster, a warning message will be displayed, but 
+  the user will be allowed to continue.
+
+* For block devices, a backing pool must be created. Only one pool can be created per set of block devices.
+
+* In the second step, user will decide whether to use an existing pool or create a new pool as the backing pool
+  for this set of RBD.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735638)
+
+* On choosing existing pool, a drop down will be displayed which will list down all the available pools in the 
+  Tendrl.
+
+* If using an existing pool, the total storage required by all the RBDs must not exceed the available capacity of  
+  the chosen cluster(i.e. Target size * #RBDs to create <= available capacity in cluster).
+
+* When user selects an existing pool, the attributed of that pool will be shown along with a graphic that reflects 
+  storage currently in use by that pool, the amount to be added, and the amount remaining in the cluster after the creation of the new RBD storage.
+
+* The details to be shown on the UI on selecting existing pool are -
+
+    - Type
+
+    - OSDs
+
+    - Replicas
+
+    - PG Count
+
+    - Journal Configuration
+
+    - Quotas
+
+* If user wants to create a new pool, they are presented with a pool creation form. See Create Pool Workflow's spec 
+  for details about these options.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735639)
+
+* User will be presented with a summary of block devices to be created, including the backing pool information.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735647)
+
+* Details on the summary page for RBDs are -
+
+    - Name
+
+    - Target Size
+
+* Details on the summary page for backing pool are -
+
+    - Name
+
+    - Type
+
+    - OSDs
+
+    - Replicas
+
+    - PG Count
+
+    - Journal Configuration
+
+    - Quotas
+
+* Clicking on "Create RBDs" button will create these RBDs and backing pool, if required.
+
+* A POST call be fired form UI to backend along with data required for creating RBD.
+
+* On successful creation, a successful message will be displayed to user.
+  (https://redhat.invisionapp.com/share/EHAC6S0N4#/screens/217735642) 
+
+== Proposed change
+
+Front-end related changes:
+
+    * Create view for "Create RBD" workflow.
+    * An API call be triggered at the bootstrap of application which will fetch the cluster list, pool list 
+      to display the 
+
+      - cluster options and size for each cluster in step one 
```

Comment:
@gnehapk @anivargi For this I feel we need pool utilization data to be attached to cluster and `Utilization.Available` need to be shown here. Will create a PR for the back-end changes as its not available at the moment. Only pool utilization data is available for individual pools.